pkg/ingester/ingester.go

```diff
@@ -474,6 +476,13 @@ func (u *userTSDB) blocksToDelete(blocks []*tsdb.Block) map[ulid.ULID]struct{} {
 		return deletable
 	}
 
+	now := time.Now().UnixMilli()
+	for _, b := range blocks {
+		if now-b.MaxTime() >= u.blockRetentionPeriod {
```

Comment:
Can we use `cfg.BlocksStorageConfig.TSDB.Retention.Milliseconds()` here instead of `u.blockRetentionPeriod`?

If we pre-initialize `u.blockRetentionPeriod` then we have to restart Ingester to get hot reload working

Reply:
So pass in the entire config into userTSDB?

Reply:
Sry we can keep your code as is. Retention is not hot reloadable